api/src/wfl/source.clj

```diff
@@ -164,7 +170,7 @@
                     (log-prefix source) (:name dataset) table begin end))
   (let [wfl   (jdbc/with-db-transaction [tx (postgres/wfl-db-config)]
                 (postgres/get-table tx details))
-        old   (when (seq wfl) (reduce combine-tdr-source-details wfl))
+        old   (filter-and-combine-tdr-source-details wfl)
```

Comment:
This handles the `nil` case.

Reply:
Yes, and I will add another test case to be explicit.